File: lib/Gaetan/news_api.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';

class NewsApi {
  
  List<Article> news  = [];

  Future<void> getNews() async{

    String url = "http://newsapi.org/v2/top-headlines?country=in&excludeDomains=stackoverflow.com&sortBy=publishedAt&language=en&apiKey=604614deb88f4ea981963e4139f50084";

    var response = await http.get(url);

    var jsonData = jsonDecode(response.body);

    if (jsonData['status'] == "ok"){
      jsonData["articles"].forEach((element){

        if (element['urlToImage'] != null && element['description'] != null){
          Article article = Article(
            title: element['title'],
            author: element['author'],
            description: element['description'],
            urlToImage: element['urlToImage'],
            publshedAt: DateTime.parse(element['publishedAt']),
            content: element["content"],
            articleUrl: element["url"],
          );
          news.add(article);
        }
        print(element['title']);
        print("\n");
      });
    }
  }
}

/*
class NewsCategory {
  List<Article> news  = [];

  Future<void> getNewsForCategory(String category) async{

    String url = "http://newsapi.org/v2/top-headlines?country=in&category=$category&apiKey=604614deb88f4ea981963e4139f50084";
    var response = await http.get(url);
    var jsonData = jsonDecode(response.body);

    if (jsonData['status'] == "ok") {
      jsonData["articles"].forEach((element) {
        if (element['urlToImage'] != null && element['description'] != null) {
          Article article = Article(
            title: element['title'],
            author: element['author'],
            description: element['description'],
            urlToImage: element['urlToImage'],
            publshedAt: DateTime.parse(element['publishedAt']),
            content: element["content"],
            articleUrl: element["url"],
          );
          news.add(article);
          list = news.news;
          list[0].urlToImage;
        }
      });
    }
  }
}*/

class Article {
  String title;
  String author;
  String description;
  String urlToImage;
  DateTime publshedAt;
  String content;
  String articleUrl;

  Article({this.title,this.description,this.author,this.content,this.publshedAt,
    this.urlToImage, this.articleUrl});
}

File: Web/public/js/newsapi.js
const news = document.getElementById('news');

// News API Data

const apiKey = '604614deb88f4ea981963e4139f50084';
const Url = 'http://newsapi.org/v2/top-headlines?country=fr&excludeDomains=stackoverflow.com&sortBy=publishedAt&language=en&apiKey=';

// Request News Function

async function getNews(url) {
  let response = await fetch(url + apiKey);
  let jsonResponse = await response.json();
  let articlesArray = jsonResponse.articles;

  articlesArray.forEach((elemnt) => {
    if (elemnt.author != null && elemnt.title != null && elemnt.description != null && elemnt.url != null && elemnt.urlToImage != null) {    
      let articleRow =
        '<div class="articlerow">' +
        ' <a href="' + elemnt.url + '" target="_blank" class="readmore"><img class="storyimage" src="' + elemnt.urlToImage + '" /></a>' +
        ' <div class="article">' +
        '   <h2 class="article_title">' + elemnt.title + '</h2>' +
        '   <p> ' + elemnt.description + '</p>' +
        '   <h3>Rédigé par ' + elemnt.author +'</h3>' +
        ' </div>' +
        '</div>';
  
      news.innerHTML += articleRow;
    }
  });
  return articlesArray;
}

async function searchButton() {
  const trump = document.getElementById("trump").value;
  
  if (trump.length == 0 || trump == "reset") {
    document.location.reload(true);
    //getNews();  le call marche, la condition est bonne mais pas d'affichage
  }
  const searchURL = "https://newsapi.org/v2/everything?q=" + trump + "&apiKey=" + apiKey;
  
  let response = await fetch(searchURL);
  let jsonResponse = await response.json();
  let articlesArray = jsonResponse.articles;

  //reset l'affichage des news
  news.innerHTML = 0;
  articlesArray.forEach((elemnt) => {
      if (elemnt.author != null && elemnt.title != null && elemnt.description != null && elemnt.url != null && elemnt.urlToImage != null) {    
        let articleRow =
          '<div class="articlerow">' +
          ' <div class="article">' +
          '   <h2 class="title">' + elemnt.title + '</h2>' +
          '   <h3>By ' + elemnt.author +'</h3>' +
          '   <p> ' + elemnt.description + '</p>' +
          '   <a href="' + elemnt.url + '" target="_blank" class="readmore">Read More</a>' +
          ' </div>' +
          ' <div class="share">' +
          '   <img class="storyimage" src="' + elemnt.urlToImage + '" />' +
          ' </div>' +
          '</div>';
    
        news.innerHTML += articleRow;
      }
    });
  return articlesArray;
}

function reset() {
  document.location.reload(true);
}

getNews(Url)